src/logging_manager/mod.rs

```diff
@@ -0,0 +1,340 @@
+    for file in read_dir(logging_directory)? {
+        let file = file?;
+
+        // check file type and name
+        if file.file_type()?.is_file() {
+            // get file name
+            let filename = match file.file_name().into_string() {
+                Ok(filename) => filename,
+                Err(_) => continue,
+            };
+
+            // check filename
+            if filename.ends_with(".log") && time_manager::check_date(&filename[0..10]) {
+                files.push(file);
+            }
+        }
+    }
+
+    // sort files by date
+    files.sort_by(|a, b| a.file_name().cmp(&b.file_name()));
+
+    Ok(files)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fs::{remove_dir_all, File, remove_dir};
+    use std::thread;
+    use std::time;
+    use std::error::Error;
+    use std::io::{BufReader, BufRead};
+    use log::*;
+
+    #[test]
+    fn test_log() -> Result<(), Box<dyn Error>> {
+        let logging_directory = Path::new("./test_logs/");
+        let logging_service = LoggingService::new(LoggingConfig {
+            logging_directory,
+            max_dir_size: 10000,
+        });
+        logging_service.init(log::LevelFilter::Trace)?;
+        let current_date = time_manager::curr_datestamp();
+        let log_file_name = format!("{}.log", current_date);
+        let log_file_path_buf = logging_directory
+            .join(log_file_name);
+        let log_file_path = log_file_path_buf.as_path();
+
+        // try logging different messages
+        // "[YYYY-MM-DD HH:MM:SS] DEBUG test debug"
+        debug!("test debug");
+        // "[YYYY-MM-DD HH:MM:SS] ERROR test error"
+        error!("test error");
+        // "[YYYY-MM-DD HH:MM:SS] INFO  test info"
+        info!("test info");
+        // "[YYYY-MM-DD HH:MM:SS] TRACE test trace"
+        trace!("test trace");
+        // "[YYYY-MM-DD HH:MM:SS] WARN  test warning"
+        warn!("test warning");
+
+        // sleep because logging is done on a different thread (and will take time)
+        thread::sleep(time::Duration::from_millis(10));
+
+        // make sure file is there and contents are correct
+        assert!(
+            log_file_path.exists()
+        );
+        let log_file = File::open(log_file_path)?;
+        let mut lines = BufReader::new(log_file).lines();
+        assert!(lines.next().unwrap().unwrap().ends_with("] DEBUG test debug"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] ERROR test error"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] INFO  test info"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] TRACE test trace"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] WARN  test warning"));
+        assert!(lines.next().is_none());
+
+        // clean up
+        remove_dir_all(logging_directory)?;
+        assert_eq!(false, logging_directory.exists());
+        Ok(())
+    }
+
+    #[test]
+    fn test_check_size() -> Result<(), std::io::Error> {
```

Comment:
Might not be a bad idea to split this test into three separate test functions.

Reply:
I prefer to have it as one

Reply:
One reason to split it up is that it's a lot easier to debug short unit tests that only test a single case each. It's more helpful to see in the test results something like "check_size_max_dir_size_0 is failing". Also, it's generally best to avoid having long functions, even if they're tests.

Reply:
Yeah those are definitely reasons lol